src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java

```diff
@@ -165,31 +167,24 @@ String getClassNameInternal() {
         stmtPoolable = true;
 
         // Create a cache key for this statement.
-        Sha1HashKey cacheKey = new Sha1HashKey(sql);
+        cacheKey = new CityHash128Key(sql);
 
         // Parse or fetch SQL metadata from cache.
-        ParsedSQLCacheItem cacheItem = getOrCreateCachedParsedSQLMetadata(cacheKey, sql);
+        ParsedSQLCacheItem parsedSql = getOrCreateCachedParsedSQL(cacheKey, sql);
 
         // Retrieve meta data from cache item.
-        procedureName = cacheItem.procedureName;
-        bReturnValueSyntax = cacheItem.bReturnValueSyntax;
-        userSQL = cacheItem.processedSQL;
-        initParams(cacheItem.parameterCount);
-
-        // See if existing prepared statement handle can be re-used.
-        cachedPreparedStatementItem = connection.borrowCachedPreparedStatementMetadata(cacheKey);
-        // If handle was found then re-use.
-        if(null != cachedPreparedStatementItem && cachedPreparedStatementItem.hasExecutedSpExecuteSql()) {
-            // If existing handle was found and we can add reference use it.
-            PreparedStatementHandle handle = cachedPreparedStatementItem.getPreparedStatementHandle();
-            if (handle.addReference()) {
-                prepStmtHandle = handle.getHandle();
-            }
-            else {
-                // Because sp_executesql was already called on this SQL-text use regular prep/exec pattern.
-                isExecutedAtLeastOnce = true;
-            }
-        }
+        procedureName = parsedSql.procedureName;
+        bReturnValueSyntax = parsedSql.bReturnValueSyntax;
+        userSQL = parsedSql.processedSQL;
+        initParams(parsedSql.parameterCount);
+
+        // Attempt to borrow the statement handle, if statement handle caching is enabled, this will be 0 if not.
+        prepStmtHandle = connection.borrowStatementHandle(cacheKey);
+        if (0 < prepStmtHandle)
```

Comment:
I was going to rebase my change onto ``mssql/metadataCaching``, but basically every change was a merge conflict, so I opted to make the spot change for this in my ``metadataCaching2`` branch.